sdk/tests/conformance2/textures/misc/copy-texture-image-luma-format.html

```diff
@@ -0,0 +1,240 @@
+    gl_Position = a_position;
+    v_coord = a_coord;
+}
+</script>
+<script id="fshader_luminance_alpha" type="x-shader/x-fragment">#version 300 es
+in highp vec2 v_coord;
+uniform highp sampler3D u_sampler0;
+uniform highp float u_depth;
+out highp vec4 o_color0;
+void main (void) {
+    o_color0 = vec4(texture(u_sampler0, vec3(v_coord, u_depth)));
+}
+</script>
+<script id="fshader_rgba_3d_alpha" type="x-shader/x-fragment">#version 300 es
+in highp vec2 v_coord;
+uniform highp sampler3D u_sampler0;
+uniform highp float u_depth;
+out highp vec4 o_color0;
+void main (void) {
+    highp float alpha = texture(u_sampler0, vec3(v_coord, u_depth)).a;
+    o_color0 = vec4(0, 0, 0, alpha);
+}
+</script>
+<script id="fshader_rgba_3d_luminance" type="x-shader/x-fragment">#version 300 es
+in highp vec2 v_coord;
+uniform highp sampler3D u_sampler0;
+uniform highp float u_depth;
+out highp vec4 o_color0;
+void main (void) {
+    highp float r = texture(u_sampler0, vec3(v_coord, u_depth)).r;
+    o_color0 = vec4(r, r, r, 1);
+}
+</script>
+<script id="fshader_rgba_3d_luminance_alpha" type="x-shader/x-fragment">#version 300 es
+in highp vec2 v_coord;
+uniform highp sampler3D u_sampler0;
+uniform highp float u_depth;
+out highp vec4 o_color0;
+void main (void) {
+    highp float r = texture(u_sampler0, vec3(v_coord, u_depth)).r;
+    highp float alpha = texture(u_sampler0, vec3(v_coord, u_depth)).a;
+    o_color0 = vec4(r, r, r, alpha);
+}
+</script>
+<script>
+"use strict";
+description("This test verifies the functionality of the coptexsubimage3d.");
+debug("");
+
+var wtu = WebGLTestUtils;
+var canvas = document.getElementById("canvas");
+var gl = wtu.create3DContext(canvas, null, 2);
+var vao = null;
+
+function copytexsubimage3D_luma_format() {
+
+    var testGroup = [
+        {
+            name: '3d_alpha',
+            format: gl.ALPHA,
+            width: 64,
+            height: 32,
+            depth: 2
+        },
+        {
+            name: '3d_luminance',
+            format: gl.LUMINANCE,
+            width: 64,
+            height: 32,
+            depth: 2
+        },
+        {
+            name: '3d_luminance_alpha',
+            format: gl.LUMINANCE_ALPHA,
+            width: 64,
+            height: 32,
+            depth: 2
+        }
+    ];
+
+    testGroup.forEach(function(testcase) {
+        debug("");
+        debug("Testing copytexsubimage3d_luma_format_" + testcase.name);
+
+        var texture = [];
+        texture[0] = gl.createTexture();
+        texture[1] = gl.createTexture();
+        texture[2] = gl.createTexture();
+        var bytes = [];
+        var layer = 0;
+        var width = testcase.width;
+        var height = testcase.height;
+        var depth = testcase.depth;
+        var w = testcase.width;
+        var h = testcase.height;
+        var format;
+        var uint = new Uint8Array(width * height * 4);
+        for (var i = 0; i < uint.length; ++i) {
+            uint[i] = Math.random() * 0xFF;
```

Comment:
It's problematic to use fully random numbers in these tests. This can cause hard-to-reproduce bugs. Please figure out a way to make this test deterministic.